scripts/changeset/get-full-changelog.mts

```diff
@@ -0,0 +1,154 @@
+import * as github from '@actions/github';
+import { getInfo } from '@changesets/get-github-info';
+import getChangesets from '@changesets/read';
+import type { NewChangeset } from '@changesets/types';
+import { execSync } from 'child_process';
+import { join } from 'path';
+
+type Octokit = ReturnType<typeof github.getOctokit>;
+
+interface ChangelogInfo {
+  isBreaking: boolean;
+  prType: string;
+  markdown: string;
+}
+
+const prTypes = ['feat', 'fix', 'refactor', 'chore', 'docs'];
+
+async function getChangelogInfo(octokit: Octokit, changeset: NewChangeset): Promise<ChangelogInfo> {
+  const changesetCommitLog = execSync(
+    `git log --oneline --diff-filter=A -- ${join(
+      process.cwd(),
+      '.changeset',
+      `${changeset.id}.md`
+    )}`
+  ).toString(); // e.g. 1f3d3d3 fix!: add breaking fix
+
+  const [commit] = changesetCommitLog.split(' ');
+
+  const {
+    links: { pull: prLink, user },
+    pull: prNo,
+  } = await getInfo({
+    repo: process.env.GITHUB_REPOSITORY ?? 'This should be set by GitHub',
+    commit,
+  });
+
+  const {
+    data: { title },
+  } = await octokit.rest.pulls.get({
+    ...github.context.repo,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    pull_number: prNo!,
+  });
+
+  const prType = title.replace(/(\w+).*/, '$1'); // chore!: add something -> chore
+  const isBreaking = title.includes(`${prType}!`);
+
+  const titleDescription = title.replace(/\w+\W+(.*)/, '$1'); // chore!: add something -> add something
```

Comment:
Should we keep the prefixes? 🤔 _(after [this](https://github.com/FuelLabs/fuels-ts/pull/1883#discussion_r1530267883))_

Reply:
There's no need to since they are part of a group:

#### Features
- [#1872 ](https://github.com/FuelLabs/fuels-ts/pull/1872) - Improve changelog template, by @nedsalk 

vs

#### Features
- [#1872 ](https://github.com/FuelLabs/fuels-ts/pull/1872) - feat: improve changelog template, by @nedsalk 

The "feat:" is redundant.

Reply:
Also, we could maybe make use of GitHub automatically printing the PR title when hashtag+pr_number isn't formatted as a link:

#### Features
- #1872, by @nedsalk

Reply:
I also prefer the cleaner one.